tasks.c

```diff
@@ -2663,22 +2670,40 @@ void vTaskStartScheduler( void )
-                StaticTask_t * pxIdleTaskTCBBuffer = NULL;
-                StackType_t * pxIdleTaskStackBuffer = NULL;
-                uint32_t ulIdleTaskStackSize;
-
-                /* The Idle task is created using user provided RAM - obtain the
-                 * address of the RAM then create the idle task. */
-                vApplicationGetIdleTaskMemory( &pxIdleTaskTCBBuffer, &pxIdleTaskStackBuffer, &ulIdleTaskStackSize );
-                xIdleTaskHandle[ xCoreID ] = xTaskCreateStatic( prvIdleTask,
-                                                                cIdleName,
-                                                                ulIdleTaskStackSize,
-                                                                ( void * ) NULL,       /*lint !e961.  The cast is not redundant for all compilers. */
-                                                                portPRIVILEGE_BIT,     /* In effect ( tskIDLE_PRIORITY | portPRIVILEGE_BIT ), but tskIDLE_PRIORITY is zero. */
-                                                                pxIdleTaskStackBuffer,
-                                                                pxIdleTaskTCBBuffer ); /*lint !e961 MISRA exception, justified as it is not a redundant explicit cast to all supported compilers. */
-
+                if(xCoreID == 0)
+                {
+                    StaticTask_t * pxIdleTaskTCBBuffer = NULL;
+                    StackType_t * pxIdleTaskStackBuffer = NULL;
+                    uint32_t ulIdleTaskStackSize;
+
+                    /* The Idle task is created using user provided RAM - obtain the
+                    * address of the RAM then create the idle task. */
+                    vApplicationGetIdleTaskMemory( &pxIdleTaskTCBBuffer, &pxIdleTaskStackBuffer, &ulIdleTaskStackSize );
+                    xIdleTaskHandle[ xCoreID ] = xTaskCreateStatic( prvIdleTask,
+                                                                    cIdleName,
+                                                                    ulIdleTaskStackSize,
+                                                                    ( void * ) NULL,       /*lint !e961.  The cast is not redundant for all compilers. */
+                                                                    portPRIVILEGE_BIT,     /* In effect ( tskIDLE_PRIORITY | portPRIVILEGE_BIT ), but tskIDLE_PRIORITY is zero. */
+                                                                    pxIdleTaskStackBuffer,
+                                                                    pxIdleTaskTCBBuffer ); /*lint !e961 MISRA exception, justified as it is not a redundant explicit cast to all supported compilers. */
+                }
+                #if( configNUM_CORES > 1)
+                else
+                {
+                    struct taskMemory{
+                        StaticTask_t TCB;
+                        StackType_t stack[configMINIMAL_STACK_SIZE];
```

Comment:
If these tasks cannot contain user code (don't call an idle hook), then I think it is ok for them all to have the same stack size (so no requested change).  Just watch the variable/struct naming convention.  We may want to look at breaking this whole function up a bit as it was already quite long, and is [obviously] longer now.  Perhaps a prvCreateIdleTasks() function to contain this part.

Reply:
prvCreateIdleTasks makes sense.  I will go ahead and do it.